packages/@aws-cdk/core/lib/tree.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Interface for inspecting a construct and adding metadata into the construct tree
+ */
+export interface ITreeInspector {
```

Comment:
Why do we need the additional interface? Doesn't feel like it adds too much value...

Reply:
We went back and forth on this one for a bit... The main rationale was extensibility where so we can add to the ITreeInspector without modifying the signature of `IInspectable`

@rix0rrr anything I'm missing here?

Reply:
I would be okay with using a concrete class. It has the same amount of extensibility. It would allow the inspectee to do some things that it shouldn't, but I guess that's probably fine.

Reply:
removed and went with a concrete class. Agree with the points about similar level of extensibility and not gaining much through the interface.